mender-client-acceptance-testing/Dockerfile

```diff
@@ -1,40 +1,52 @@
-FROM ubuntu:22.04
+FROM python:3.12.5-slim-bookworm AS python-source
+
+FROM ubuntu:24.04
 
 ENV DEBIAN_FRONTEND=noninteractive
 
 # prepare NFS cache for yocto
 RUN mkdir -p /mnt/sstate-cache
 
-# install pyyaml via pip to workardound a later conflict if installed via OS package (awscli depends on python3-yaml)
-RUN apt-get update -qq && apt-get install -yyq python3-pip && pip3 install --upgrade pip && pip3 install pyyaml
-
-# Get OS requirements from master
-# linux-modules-$(uname -r | sed 's/gcp/gke/') is a workaround for https://northerntech.atlassian.net/browse/QA-597
-RUN apt-get install -yyq wget && \
+# OS requirements from master
+RUN apt-get update -qq && apt-get install -yyq wget && \
     wget https://raw.githubusercontent.com/mendersoftware/meta-mender/master/tests/acceptance/requirements_deb.txt && \
-    apt-get install -yyq $(cat requirements_deb.txt) && apt-get install -yq linux-modules-$(uname -r | sed 's/gcp/gke/') && \
-    apt-get remove -yyq docker docker.io containerd runc && apt-get install -yyq ca-certificates curl gnupg lsb-release && \
-    mkdir -p /etc/apt/keyrings && curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg && \
-    echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null && \
-    apt-get update -qq && apt-get install -yyq docker-ce docker-ce-cli containerd.io docker-compose-plugin
+    apt-get install -yyq $(cat requirements_deb.txt) && \
+    rm requirements_deb.txt
+
+# Install kernel modules for KVM. Replacing 's/gcp/gke/') is a workaround for QA-597.
+RUN apt-get install -yq linux-modules-$(uname -r | sed 's/gcp/gke/')
+
+# Install Docker Engine as per official instructions. See https://docs.docker.com/engine/install/ubuntu/
+RUN for pkg in docker.io docker-doc docker-compose docker-compose-v2 podman-docker containerd runc; do \
+        apt-get remove -yyq $pkg; done && \
+    apt-get update -qq && \
+    apt-get install -yyq ca-certificates curl  && \
+    install -m 0755 -d /etc/apt/keyrings  && \
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc  && \
+    chmod a+r /etc/apt/keyrings/docker.asc && \
+    echo \
+        "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+        $(. /etc/os-release && echo "$VERSION_CODENAME") stable" > /etc/apt/sources.list.d/docker.list && \
+    apt-get update -qq && \
+    apt-get install -yyq docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+
+# Python requirements from master
+# Use a local instalation of Python to avoid collisions with Debian native packages. See PEP 668.
+COPY --from=python-source /usr/local /usr/local
```

Comment:
Here my only concern would be if any of the prior installed packages might have something in `/usr/local` that would be overwritten here and then break...

Reply:
It should not be the case, `/usr/local` is for *user* locally installed software, and the OS packages should go into `/usr`. At least official Debian packages will respect this for sure, maybe third party (Docker, in this case) can mess it up but I don't think so as it is quite a well established policy.

The base image comes with `unminimize` binary in there, but we don't need that in our image.

Reply:
I had the same concern, but since you seems confident it should not break (or at least there are very tiny chances to break it), then LGTM.

Reply:
Ah ok, so it's just the python installation that is put there then 👌